File: SendRecv/basic.f90
PROGRAM basic

  IMPLICIT NONE
  INCLUDE 'mpif.h'
  INTEGER,PARAMETER::size=100
  INTEGER::ierror, myid,ntasks,count
  INTEGER:: status(MPI_STATUS_SIZE)
  INTEGER:: message(size)
  
  CALL MPI_INIT(ierror)
  CALL MPI_COMM_RANK(MPI_COMM_WORLD, myid, ierror)
  CALL MPI_COMM_SIZE(MPI_COMM_WORLD, ntasks, ierror)

  message = myid

!Send and receive as defined in exercise

  CALL MPI_FINALIZE(ierror)

END PROGRAM basic

File: SendRecv/basic.c
#include<stdio.h>
#include<stdlib.h>
#include<mpi.h>

int main(int argc,char *argv[])
{
    int i,myid, ntasks;
    int size=1000000;
    int *message;
    MPI_Status status;
    MPI_Init(&argc,&argv);
    MPI_Comm_size(MPI_COMM_WORLD,&ntasks);
    MPI_Comm_rank(MPI_COMM_WORLD,&myid);
    
    //allocate message 
    message = malloc(sizeof(int)*size);
    //initialize message
    for(i=0;i<size;i++) message[i]=myid;
    
    //send and receive messages as defined in exercise
    
    free(message);
    
    MPI_Finalize();
}
